src/mono/mono/component/debugger-agent.c

```diff
@@ -10202,22 +10265,53 @@ debugger_thread (void *arg)
                 if (mono_metadata_has_updates_api ()) {
                         PRINT_DEBUG_MSG (1, "[dbg] Cannot attach after System.Reflection.Metadata.MetadataUpdater.ApplyChanges has been called.\n");
                         attach_failed = TRUE;
-                        command_set = (CommandSet)0;
-                        command = 0;
                         dispose_vm ();
```

Comment:
Should `dispose_vm` here mean that `is_vm_dispose_command=TRUE`, so it affects the condition on line 10283?
`	if (!attach_failed && is_vm_dispose_command && !(vm_death_event_sent || mono_runtime_is_shutting_down ())) {`